plugins/jobs/src/js/types/Job.ts

```diff
@@ -103,73 +107,99 @@ export function JobTypeResolver(response: MetronomeJobDetailResponse): Job {
     schedules: JobFieldResolvers.schedules(response)
   };
 }
+
 export const JobFieldResolvers = {
-  id(job: MetronomeJobDetailResponse): string {
+  id(job: MetronomeJobDetailResponse | MetronomeJobResponse): string {
     return job.id;
   },
-  description(job: MetronomeJobDetailResponse): string {
-    return job.description;
+  description(
+    job: MetronomeJobDetailResponse | MetronomeJobResponse
+  ): string | null {
+    return isMetronomeJobDetailResponse(job) ? job.description : null;
   },
-  command(job: MetronomeJobDetailResponse): string {
+  command(job: MetronomeJobDetailResponse | MetronomeJobResponse): string {
     return job.run.cmd;
   },
-  disk(job: MetronomeJobDetailResponse): number {
+  disk(job: MetronomeJobDetailResponse | MetronomeJobResponse): number {
     return job.run.disk;
   },
-  mem(job: MetronomeJobDetailResponse): number {
+  mem(job: MetronomeJobDetailResponse | MetronomeJobResponse): number {
     return job.run.mem;
   },
-  cpus(job: MetronomeJobDetailResponse): number {
+  cpus(job: MetronomeJobDetailResponse | MetronomeJobResponse): number {
     return job.run.cpus;
   },
-  activeRuns(job: MetronomeJobDetailResponse): JobRunConnection {
-    return JobRunConnectionTypeResolver(job.activeRuns);
+  activeRuns(
+    job: MetronomeJobDetailResponse | MetronomeJobResponse
+  ): JobRunConnection | null {
+    return isMetronomeJobDetailResponse(job)
+      ? JobRunConnectionTypeResolver(job.activeRuns)
+      : null;
   },
-  docker(job: MetronomeJobDetailResponse): JobDocker | null {
-    return job.run.docker ? JobDockerTypeResolver(job.run.docker) : null;
+  docker(
+    job: MetronomeJobDetailResponse | MetronomeJobResponse
+  ): JobDocker | null {
+    if (isMetronomeJobDetailResponse(job) && job.run.docker) {
+      return JobDockerTypeResolver(job.run.docker);
+    }
+    return null;
   },
-  jobRuns(job: MetronomeJobDetailResponse): JobRunConnection {
-    return AddStatusToHistoryJobRuns(job);
+  jobRuns(
+    job: MetronomeJobDetailResponse | MetronomeJobResponse
+  ): JobRunConnection | null {
+    return isMetronomeJobDetailResponse(job)
+      ? AddStatusToHistoryJobRuns(job)
+      : null;
   },
-  json(job: MetronomeJobDetailResponse): string {
+  json(job: MetronomeJobDetailResponse | MetronomeJobResponse): string {
     return JSON.stringify(cleanJobJSON(job));
   },
-  labels(job: MetronomeJobDetailResponse): JobLabel[] {
+  labels(job: MetronomeJobDetailResponse | MetronomeJobResponse): JobLabel[] {
     return Object.entries(job.labels).map(([key, value]) => ({ key, value }));
   },
-  lastRunsSummary(job: MetronomeJobDetailResponse): JobHistorySummary {
-    return JobHistorySummaryTypeResolver(job.history);
+  lastRunsSummary(
+    job: MetronomeJobDetailResponse | MetronomeJobResponse
+  ): JobHistorySummary | null {
+    return isMetronomeJobDetailResponse(job)
+      ? JobHistorySummaryTypeResolver(job.history)
```

Comment:
this needs a small fixup, its `job.history` but `jobs[].historySummary` - content is almost the same tho

Reply:
let's fix this